packages/entrykit/src/onboarding/Session.tsx

```diff
@@ -1,30 +1,52 @@
+import { useEffect } from "react";
+import { Hex } from "viem";
+import { useBalance, useWatchBlockNumber } from "wagmi";
 import { Button } from "../ui/Button";
 import { useSetupSession } from "./useSetupSession";
 import { ConnectedClient } from "../common";
-import { useEffect } from "react";
 import { useSessionClient } from "../useSessionClient";
 import { useShowQueryError } from "../errors/useShowQueryError";
 import { useShowMutationError } from "../errors/useShowMutationError";
 import { StepContentProps } from "./common";
+import { useEntryKitConfig } from "../EntryKitConfigProvider";
 
 export type Props = StepContentProps & {
   userClient: ConnectedClient;
   registerSpender: boolean;
   registerDelegation: boolean;
+  sessionAddress?: Hex;
 };
 
-export function Session({ isActive, isExpanded, userClient, registerSpender, registerDelegation }: Props) {
+export function Session({
+  isActive,
+  isExpanded,
+  userClient,
+  registerSpender,
+  registerDelegation,
+  sessionAddress,
+}: Props) {
   const sessionClient = useShowQueryError(useSessionClient(userClient.account.address));
   const setup = useShowMutationError(useSetupSession({ userClient }));
   const hasSession = !registerDelegation && !registerDelegation;
 
+  const { chain } = useEntryKitConfig();
+  const balance = useShowQueryError(useBalance({ chainId: chain.id, address: sessionAddress }));
+  useWatchBlockNumber({ onBlockNumber: () => balance.refetch() });
+
   useEffect(() => {
     // There seems to be a tanstack-query bug(?) where multiple simultaneous renders loses
     // state between the two mutations. They're not treated as shared state but rather
     // individual mutations, even though the keys match. And the one we want the status of
     // seems to stay pending. This is sorta resolved by triggering this after a timeout.
     const timer = setTimeout(() => {
-      if (isActive && setup.status === "idle" && sessionClient.data && !hasSession) {
+      if (
+        isActive &&
+        setup.status === "idle" &&
+        sessionClient.data &&
+        !hasSession &&
+        balance.data?.value != null &&
+        balance.data.value > 0n
```

Comment:
In the "gas tank" variant, the session account may not have a balance, because we first need to register it as a spender for the main account. If i understand this correctly it would only trigger `registerSpender` after the session wallet has an own balance?

Reply:
Exactly, once it has own balance, it attempts to `registerSpender()`. Without this condition, it tries to register it even before it has own balance which is what causes the "Enable" button pending state prematurely.